tests/helpers/kube.go

```diff
@@ -276,6 +286,40 @@ func WaitForNodeEvent(ctx context.Context, t *testing.T, c klient.Client, nodeNa
 	}, WaitTimeout, WaitInterval, "node %s should have event %v", nodeName, expectedEvent)
 }
 
+// SelectTestNodeFromUnusedPool selects a test node from the second half of the cluster's node list.
+// This helps avoid collisions with other tests that use the first half.
+// Prefers uncordoned nodes but will fall back to the first node in the pool if none are available.
+func SelectTestNodeFromUnusedPool(ctx context.Context, t *testing.T, client klient.Client) string {
```

Comment:
Why do we need this? Are there tests that are not cleaning up state correctly?

Reply:
Picking only schedulable nodes now. This because we've had problem in past where one test failure was leading to cascade of failure as test was picking the same node again and again